R/rromeo_publishers.R

```diff
@@ -0,0 +1,252 @@
+                           key = NULL) {
+
+  romeo_colour = match.arg(romeo_colour)
+
+  api_answer = GET(rr_base_api(), query = list(colour = romeo_colour,
+                                               ak = check_key(key)))
+
+  parse_publisher(api_answer)
+}
+
+
+#' Get Publisher Policy by Publisher Name
+#'
+#' Use SHERPA/RoMEO API to retrieve a specific publisher policies on manuscript
+#' archival based on matching the name of the publishers.
+#'
+#' @param name  {`character(1+)`}\cr{}
+#'              One or a vector of query string(s) to search publisher name
+#' @param qtype {`character(1)`}\cr{}
+#'              in `c("all", "any", "exact")` define the type of matching:
+#'              * `all` means that all strings in `name` must appear in any
+#'                order or location
+#'              * `any` means that at least one of the strings in `name` must
+#'                appear
+#'              * `exact` means that the `name` string must appear in the
+#'                publisher's name or its alias.
+#'
+#' @inheritParams check_key
+#'
+#' @inherit parse_publisher return
+#'
+#' @inherit check_key details
+#'
+#' @importFrom httr GET
+#' @export
+#'
+#' @examples
+#'
+#' rr_publisher_name(name = "Optical Society", qtype = "all")
+#' rr_publisher_name(name = "Swiss Chemistry", qtype = "any")
+#' rr_publisher_name(name = "Swiss Chemistry", qtype = "exact")
+rr_publisher_name = function(name, qtype = c("all", "any", "exact"),
+                             key = NULL) {
+
+  qtype = match.arg(qtype)
+
+  api_key = check_key(key)
+
+  answer_list = lapply(name, function(publisher_name) {
+    api_answer = GET(rr_base_api(), query = list(pub   = publisher_name,
+                                                 qtype = qtype,
+                                                 ak    = api_key))
+
+    parse_publisher(api_answer)
+  })
+
+  publishers_df = do.call(rbind.data.frame,
+                          c(answer_list, stringsAsFactors = FALSE))
+
+  return(publishers_df)
+}
+
+#' Get Publisher Policy by Publisher's Continent
+#'
+#' Retrieve publisher's policy based on publisher's continent. This function
+#' does not work for unclassified or international publishers.
+#'
+#' @param continent {`character(1)`}\cr{}
+#'                  in ```
+#'                  c("Africa", "Antarctica",  "Asia",  "Australasia",
+#'                  "Carribean",  "Central America",  "Europe",
+#'                  "North America",  "Oceania",  "South America")
+#'                  ```\cr{}
+#'                  the continent name to retrieve
+#' @inheritParams check_key
+#'
+#' @inherit parse_publisher return
+#'
+#' @inherit check_key details
+#'
+#' @importFrom httr GET
+#' @export
+#'
+#' @examples
+#'
+#' rr_publisher_continent(continent = "Caribbean")
+#' rr_publisher_continent(continent = "Central America")
+rr_publisher_continent = function(continent = c("Africa",
```

Comment:
Why not allow the user to query multiple continent at once? We're doing it for every other function.